File: binding/src/libqfontinfo.cpp
#include <QFont>
#include <QFontInfo>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <qfontinfo.h>
#include "libqfontinfo.h"
#include "libqfontinfo.hxx"

QFontInfo* QFontInfo_new(const QFont* param1) {
	 return new QFontInfo(*param1);
}

QFontInfo* QFontInfo_new2(const QFontInfo* param1) {
	 return new QFontInfo(*param1);
}

void QFontInfo_OperatorAssign(QFontInfo* self, const QFontInfo* param1) {
	self->operator=(*param1);
}

void QFontInfo_Swap(QFontInfo* self, QFontInfo* other) {
	self->swap(*other);
}

libqt_string QFontInfo_Family(const QFontInfo* self) {
	QString _ret = self->family();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QFontInfo_StyleName(const QFontInfo* self) {
	QString _ret = self->styleName();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

int QFontInfo_PixelSize(const QFontInfo* self) {
	return self->pixelSize();
}

int QFontInfo_PointSize(const QFontInfo* self) {
	return self->pointSize();
}

double QFontInfo_PointSizeF(const QFontInfo* self) {
	return self->pointSizeF();
}

bool QFontInfo_Italic(const QFontInfo* self) {
	return self->italic();
}

int QFontInfo_Style(const QFontInfo* self) {
	return self->style();
}

int QFontInfo_Weight(const QFontInfo* self) {
	return self->weight();
}

bool QFontInfo_Bold(const QFontInfo* self) {
	return self->bold();
}

bool QFontInfo_Underline(const QFontInfo* self) {
	return self->underline();
}

bool QFontInfo_Overline(const QFontInfo* self) {
	return self->overline();
}

bool QFontInfo_StrikeOut(const QFontInfo* self) {
	return self->strikeOut();
}

bool QFontInfo_FixedPitch(const QFontInfo* self) {
	return self->fixedPitch();
}

int QFontInfo_StyleHint(const QFontInfo* self) {
	return self->styleHint();
}

int QFontInfo_LegacyWeight(const QFontInfo* self) {
	return self->legacyWeight();
}

bool QFontInfo_ExactMatch(const QFontInfo* self) {
	return self->exactMatch();
}

void QFontInfo_Delete(QFontInfo* self) {
    delete self;
}

File: binding/include/libqfontinfo.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qfontinfo.html
pub const qfontinfo = struct {

    /// New constructs a new QFontInfo object.
    pub fn New(param1: ?*anyopaque) QtC.QFontInfo {
        return qtc.QFontInfo_new(@ptrCast(param1));
    }


    /// New2 constructs a new QFontInfo object.
    pub fn New2(param1: ?*anyopaque) QtC.QFontInfo {
        return qtc.QFontInfo_new2(@ptrCast(param1));
    }


    pub fn OperatorAssign(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QFontInfo_OperatorAssign(@ptrCast(self), @ptrCast(param1));
    }

    pub fn Swap(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QFontInfo_Swap(@ptrCast(self), @ptrCast(other));
    }

    pub fn Family(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QFontInfo_Family(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qfontinfo.Family: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn StyleName(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QFontInfo_StyleName(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qfontinfo.StyleName: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn PixelSize(self: ?*anyopaque, ) i32 {
        return qtc.QFontInfo_PixelSize(@ptrCast(self));
    }

    pub fn PointSize(self: ?*anyopaque, ) i32 {
        return qtc.QFontInfo_PointSize(@ptrCast(self));
    }

    pub fn PointSizeF(self: ?*anyopaque, ) f64 {
        return qtc.QFontInfo_PointSizeF(@ptrCast(self));
    }

    pub fn Italic(self: ?*anyopaque, ) bool {
        return qtc.QFontInfo_Italic(@ptrCast(self));
    }

    pub fn Style(self: ?*anyopaque, ) i32 {
        return qtc.QFontInfo_Style(@ptrCast(self));
    }

    pub fn Weight(self: ?*anyopaque, ) i32 {
        return qtc.QFontInfo_Weight(@ptrCast(self));
    }

    pub fn Bold(self: ?*anyopaque, ) bool {
        return qtc.QFontInfo_Bold(@ptrCast(self));
    }

    pub fn Underline(self: ?*anyopaque, ) bool {
        return qtc.QFontInfo_Underline(@ptrCast(self));
    }

    pub fn Overline(self: ?*anyopaque, ) bool {
        return qtc.QFontInfo_Overline(@ptrCast(self));
    }

    pub fn StrikeOut(self: ?*anyopaque, ) bool {
        return qtc.QFontInfo_StrikeOut(@ptrCast(self));
    }

    pub fn FixedPitch(self: ?*anyopaque, ) bool {
        return qtc.QFontInfo_FixedPitch(@ptrCast(self));
    }

    pub fn StyleHint(self: ?*anyopaque, ) i32 {
        return qtc.QFontInfo_StyleHint(@ptrCast(self));
    }

    pub fn LegacyWeight(self: ?*anyopaque, ) i32 {
        return qtc.QFontInfo_LegacyWeight(@ptrCast(self));
    }

    pub fn ExactMatch(self: ?*anyopaque, ) bool {
        return qtc.QFontInfo_ExactMatch(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QFontInfo_Delete(@ptrCast(self));
    }
};
